src/domain/notification/aggregate.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable max-lines-per-function */
+import type { StoreApi } from 'zustand'
+import { immer } from 'zustand/middleware/immer'
+import { createStore } from 'zustand/vanilla'
+import { devtools } from 'zustand/middleware'
+import type { IO } from 'fp-ts/lib/IO'
+import type { Reader } from 'fp-ts/lib/Reader'
+import * as A from 'fp-ts/lib/Array'
+import * as O from 'fp-ts/lib/Option'
+import { pipe } from 'fp-ts/lib/function'
+import type { ForgetType } from '@/util/type'
+import { eqNotification } from './entity'
+import type { NotificationID, NotificationType, Notifications } from './entity'
+
+type NotificationState<T> = {
+  aggregate: Notifications<T>
+}
+
+export type ReportNotificationInput<T> = {
+  id: NotificationID
+  type: NotificationType
+  title: string
+  message?: string
+  action?: T
+}
+
+export type DismissNotificationInput = {
+  id: NotificationID
+}
+
+type NotificationAction<T> = {
+  reportNotification: (input: ReportNotificationInput<T>) => IO<void>
+  dismissNotification: (input: DismissNotificationInput) => IO<void>
+  notificationWithActionReported: () => IO<boolean>
+}
+
+export type NotificationDTO<T> = {
+  id: NotificationID
+  type: NotificationType
+  title: string
+  message?: string
+  action?: T
+}
+
+export type NotificationsDTO<T> = NotificationDTO<T>[]
```

Comment:
Maybe it would have been more readable to move DTO types into a dedicated `dto.ts` file

Reply:
IMHO, it's best to avoid the use of DTOs in the long term. A request/response approach specific to APIs should be adopted, as is the case here, distinguishing between what is `input` and what is `output`, with different types for each interaction.

Reply:
and rename the file `aggregate.ts` to `domain.ts`?

Reply:
@ccamel 
Yes, I agree.
`aggregate` looks too much confusing to my eyes.

Reply:
Another approach linked to DDD structures could consist in naming the file with its purpose, like `notification.ts`